### ISSUE TITLE
vim hang (~10 s), StackOverflowError on stdout of REPL

### ISSUE THREAD

**Issue by overthink** [tpope/vim-fireplace#249]
Repro:
- clone https://github.com/overthink/fireplace-bug
- `lein repl`
- observe expected error in lein with `(require 'fireplace-bug.core :reload)`
- open src/fireplace-bug/core.clj in vim
- `:Require`

Expected: see the expected error `CompilerException clojure.lang.ExceptionInfo: Input to h does not match schema ...` somewhere in vim.

Actual: vim hangs for ~10s, eventually see the following on lein REPL:

```
ERROR: Unhandled REPL handler exception processing message {:id fireplace-mark-pc2-1448982515-27, :op stacktrace, :session 30958f87-52fc-4259-9d23-ee1f55d3726e
}
java.lang.StackOverflowError
        at clojure.lang.PersistentArrayMap.assoc(PersistentArrayMap.java:189)
        at clojure.lang.PersistentArrayMap.assoc(PersistentArrayMap.java:29)
        at clojure.lang.RT.assoc(RT.java:778)
        at clojure.core$assoc__4142.invoke(core.clj:191)
        at clojure.lang.AFn.applyToHelper(AFn.java:160)
        at clojure.lang.RestFn.applyTo(RestFn.java:132)
        at clojure.lang.Ref.alter(Ref.java:171)
        at clojure.core$alter.doInvoke(core.clj:2322)
        at clojure.lang.RestFn.invoke(RestFn.java:464)
        at clojure.pprint$set_field.invoke(column_writer.clj:29)
        at clojure.pprint$column_writer$fn__7911$fn__7915.invoke(column_writer.clj:73)
        at clojure.lang.AFn.call(AFn.java:18)
        at clojure.lang.LockingTransaction.runInTransaction(LockingTransaction.java:235)
        at clojure.pprint$column_writer$fn__7911.invoke(column_writer.clj:72)
        at clojure.pprint.proxy$java.io.Writer$IDeref$e595a7c1.write(Unknown Source)
        at sun.reflect.GeneratedMethodAccessor11.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:606)
        at clojure.lang.Reflector.invokeMatchingMethod(Reflector.java:93)
        at clojure.lang.Reflector.invokeInstanceMethod(Reflector.java:28)
        at clojure.pprint$fn__7964.invoke(pretty_writer.clj:128)
        at clojure.lang.MultiFn.invoke(MultiFn.java:233)
        at clojure.pprint$write_tokens.invoke(pretty_writer.clj:146)
        at clojure.pprint$write_token_string.invoke(pretty_writer.clj:260)
        at clojure.pprint$write_line$fn__8042.invoke(pretty_writer.clj:293)
        at clojure.lang.AFn.call(AFn.java:18)
        at clojure.lang.LockingTransaction.runInTransaction(LockingTransaction.java:235)
        at clojure.pprint$write_line.invoke(pretty_writer.clj:288)
        at clojure.pprint$add_to_buffer$fn__8045.invoke(pretty_writer.clj:305)
        at clojure.lang.AFn.call(AFn.java:18)
        at clojure.lang.LockingTransaction.runInTransaction(LockingTransaction.java:235)
        at clojure.pprint$add_to_buffer.invoke(pretty_writer.clj:302)
        at clojure.pprint$pretty_writer$fn__8068$fn__8072.invoke(pretty_writer.clj:404)
        at clojure.lang.AFn.call(AFn.java:18)
        at clojure.lang.LockingTransaction.run(LockingTransaction.java:273)
        at clojure.lang.LockingTransaction.runInTransaction(LockingTransaction.java:229)
        at clojure.pprint$pretty_writer$fn__8068.invoke(pretty_writer.clj:395)
        at clojure.pprint.proxy$java.io.Writer$IDeref$PrettyFlush$4923d848.write(Unknown Source)
        at clojure.core$print_simple.invoke(core_print.clj:78)
        at clojure.core$fn__5831.invoke(core_print.clj:143)
        at clojure.lang.MultiFn.invoke(MultiFn.java:233)
        at clojure.core$pr_on.invoke(core.clj:3548)
        at clojure.core$pr.invoke(core.clj:3560)
        at clojure.pprint$pprint_simple_default.invoke(dispatch.clj:144)
        at clojure.lang.MultiFn.invoke(MultiFn.java:229)
        at clojure.pprint$write_out.invoke(pprint_base.clj:194)
        at clojure.pprint$pprint_vector$fn__8736.invoke(dispatch.clj:83)
        at clojure.pprint$pprint_vector.invoke(dispatch.clj:82)
repeat
        at clojure.lang.MultiFn.invoke(MultiFn.java:229)
        at clojure.pprint$write_out.invoke(pprint_base.clj:194)
        at clojure.pprint$pprint_simple_list$fn__8730.invoke(dispatch.clj:68)
        at clojure.pprint$pprint_simple_list.invoke(dispatch.clj:67)
        at clojure.pprint$pprint_list.invoke(dispatch.clj:78)
/repeat
```

**Comment by kendagriff** [comment]
This is now happening for me in an empty lein project with Clojure 1.10.0 only. Haven't found a solution yet...

**Comment by kendagriff** [comment]
I *think* I resolved this by updating to lein 2.8.3 (fixed it in test projects, still working on our main app). But I'm fairly confident vim-fireplace falls down on earlier versions of lein combined with 1.10.0.

**Comment by tpope** [comment]
Appears to be working for me now too, crossing my fingers this is fixed.
